apis/projectcontour/v1/httpproxy.go

```diff
@@ -294,6 +297,20 @@ type LoadBalancerPolicy struct {
 	Strategy string `json:"strategy,omitempty"`
 }
 
+// HeaderPolicy defines how headers are managed during forwarding
+type HeaderPolicy struct {
+	// ReplaceHeader defines how headers will be replaced during forwarding
+	ReplaceHeader []ReplaceHeader `json:"replaceHeader,omitempty"`
```

Comment:
crd question, this is not marked +optional, does that mean that the validation will enforce that iff `headerPolicy` is present, there must be at least one `replaceHeader` entry?

Reply:
Should we pluralize to `ReplaceHeaders`?

Reply:
@davecheney I found in testing that the optional doesn't really matter. I added an empty `headerPolicy`, then a `headerPolicy` with empty`replaceHeaders` and there weren't any validation errors. Let me look to see if I can get proper errors in there.

Reply:
@jpeach I think that's a good idea. Matches what we did before with `Conditions`, etc.